src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+
+Instead, we want each component instance to only manage its own data. For that to happen,each instance must generate a unique data object. In JavaScript, this can be accomplished by returning the object in a function:
+
+``` js
+data: function () {
+  return {
+    listTitle: '',
+    todos: []
+  }
+}
+```
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+Vue.component('some-comp', {
+  data: {
+    foo: 'bar'
+  }
+})
+```
+
+``` js
+export default {
+  data: {
+    foo: 'bar'
+  }
+}
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+``` js
+Vue.component('some-comp', {
+  data: function () {
+    return {
+      foo: 'bar'
+    }
+  }
+})
+```
+
+``` js
+// In a .vue file
+export default {
+  data () {
+    return {
+      foo: 'bar'
+    }
+  }
+}
+```
+
+``` js
+// It's OK to use an object directly in a root
+// Vue instance, since only a single instance
+// will ever exist.
+new Vue({
+  data: {
+    foo: 'bar'
+  }
+})
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Prop definitions <sup data-p="a">essential</sup>
+
+**Prop definitions should be as detailed as possible.**
+
+In committed code, prop definitions should always be as detailed as possible, specifying at least type(s).
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+Detailed [prop definitions](https://vuejs.org/v2/guide/components.html#Prop-Validation) have two advantages:
+
+- They document the API of the component, so that it's easy to see how the component is meant to be used.
+- In development, Vue will warn you if a component is ever provided incorrectly formatted props, helping you catch potential sources of error.
+
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+// This is only OK when prototyping
+props: ['status']
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+props: {
+  status: String
+}
+```
+
+``` js
+// Even better!
+props: {
+  status: {
+    type: String,
+    required: true,
+    validate: function (value) {
+      return [
+        'syncing',
+        'synced',
+        'version-conflict',
+        'error'
+      ].indexOf(value) !== -1
+    }
+  }
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Keyed `v-for` <sup data-p="a">essential</sup>
+
+**Always use `key` with `v-for`.**
+
+It's _always_ required on components, in order to maintain internal component state down the subtree. Even for elements though, it's a good practice to maintain predictable behavior, such as [object constancy](https://bost.ocks.org/mike/constancy/) in animations.
```

Comment:
This sentence reads a bit weird to me. What's always required on components again?

Reply:
Always using `key` with `v-for` is always required on components.

Reply:
I got it, but wouldn't it read more natural if we revise the section like this?

```md
**Always use `key` with `v-for`.**
 
`key` with `v-for` is _always_ required on components, in order to maintain internal component state down the subtree. Even for elements though, it's a good practice to maintain predictable behavior, such as [object constancy](https://bost.ocks.org/mike/constancy/) in animations.
```

Since the previous sentence is actually a heading, I guess we can/should repeat the subject in the explaining paragraph.

Reply:
Sure, I don't mind updating.